<img src="images/usm.jpg" width="480" height="240" align="left"/>

# MAT281 - Laboratorio N°03

## Objetivos de la clase

* Reforzar los conceptos básicos de pandas.

## Contenidos

* [Problema 01](#p1)


## Problema 01


<img src="https://imagenes.universia.net/gc/net/images/practicas-empleo/p/pr/pro/profesiones-con-el-avance-de-la-tecnologia.jpg" width="480" height="360" align="center"/>


EL conjunto de datos se denomina `ocupation.csv`, el cual contiene información tal como: edad ,sexo, profesión, etc.

Lo primero es cargar el conjunto de datos y ver las primeras filas que lo componen:

In [1]:
import pandas as pd
import os

In [2]:
# cargar datos
df = pd.read_csv(os.path.join("data_lab_03","ocupation.csv"), sep="|").set_index('user_id')
df.head()

age gender  occupation zip_code
user_id                                 
1         24      M  technician    85711
2         53      F       other    94043
3         23      M      writer    32067
4         24      M  technician    43537
5         33      F       other    15213

El objetivo es tratar de obtener la mayor información posible de este conjunto de datos. Para cumplir este objetivo debe resolver las siguientes problemáticas:

1. ¿Cuál es el número de observaciones en el conjunto de datos?

In [3]:
df.shape

(943, 4)

Se puede ver que las dimensiones del dataframe son ***(943,4)*** con lo que tenemos ***943*** observaciones.

2. ¿Cuál es el número de columnas en el conjunto de datos?

Por lo dicho anteriormente, son ***4*** columnas.

3. Imprime el nombre de todas las columnas

Se tiene que el nombre de todas las columnas son:

In [4]:
df.columns

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

Por otro lado, tambien se puede ver de que tipo es cada columna:

In [5]:
df.dtypes

age            int64
gender        object
occupation    object
zip_code      object
dtype: object

4. Imprima el índice del dataframe

Como se pudo ver al cargar los datos del archivo, el indice del archivo es *user_id* que consiste en numeros enteros.

In [6]:
df.index

Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)

5. ¿Cuál es el tipo de datos de cada columna?

Aquí se puede ver que los tipos de datos de cada columna son:

In [7]:
df.dtypes

age            int64
gender        object
occupation    object
zip_code      object
dtype: object

6. Resumir el conjunto de datos

In [8]:
df.describe()

age
count  943.000000
mean    34.051962
std     12.192740
min      7.000000
25%     25.000000
50%     31.000000
75%     43.000000
max     73.000000

Aquí se puede ver un resumen de los datos numericos que hay en el dataframe.

7. Resume conjunto de datos  con todas las columnas

In [9]:
df.describe(include='all')

age gender occupation zip_code
count   943.000000    943        943      943
unique         NaN      2         21      795
top            NaN      M    student    55414
freq           NaN    670        196        9
mean     34.051962    NaN        NaN      NaN
std      12.192740    NaN        NaN      NaN
min       7.000000    NaN        NaN      NaN
25%      25.000000    NaN        NaN      NaN
50%      31.000000    NaN        NaN      NaN
75%      43.000000    NaN        NaN      NaN
max      73.000000    NaN        NaN      NaN

Se puede ver que para los datos no numericos, se presentan cuantas distintas "cualidades" hay por columna, un ejemplo es decir que en *gender* hay 2, las cuales son **M** y **F**.

Y tambien muestra la frecuencia, y la moda de cada una.

8. Imprimir solo la columna de **occupation**.

In [10]:
df["occupation"]

user_id
1         technician
2              other
3             writer
4         technician
5              other
           ...      
939          student
940    administrator
941          student
942        librarian
943          student
Name: occupation, Length: 943, dtype: object

Acá se puede ver la columna **occupation** con su respectivo *user_id*.

9. ¿Cuántas ocupaciones diferentes hay en este conjunto de datos?

In [11]:
len(df["occupation"].unique())

21

Se puede ver que hay **21** ocupaciones diferentes, las cuales son:

In [12]:
df["occupation"].unique()

array(['technician', 'other', 'writer', 'executive', 'administrator',
       'student', 'lawyer', 'educator', 'scientist', 'entertainment',
       'programmer', 'librarian', 'homemaker', 'artist', 'engineer',
       'marketing', 'none', 'healthcare', 'retired', 'salesman', 'doctor'],
      dtype=object)

10. ¿Cuál es la ocupación más frecuente?

In [13]:
ocupaciones = df['occupation'].unique()
L, numero = [],0
for ocupacion in ocupaciones:
    df_aux = df.loc[lambda x: x['occupation'] == ocupacion]
    if numero < len(df_aux):
        numero = len(df_aux)
        L.append(ocupacion)
L[-1]

'student'

Se tiene que la ocupacion mas frecuente es ***student***.

11. ¿Cuál es la edad media de los usuarios?

In [14]:
df['age'].unique().mean()

40.01639344262295

12. ¿Cuál es la edad con menos ocurrencia?

In [15]:
Edades = df['age'].unique()
Ocurrencia, Lista1, Lista2,Lf = df.shape[0],[],[],[]
for edad in Edades:
    df_aux = df.loc[lambda x: x["age"] == edad]
    if Ocurrencia >= len(df_aux):
        Ocurrencia = len(df_aux)
        Lista1.append(edad)
        Lista2.append(Ocurrencia)
numero, largo = min(Lista2),len(Lista2)
for i in range(0,largo):
    if Lista2[i] == numero:
        Lf.append(Lista1[i])
Lf

[7, 66, 11, 10, 73]

Luego, tenemos que estas son las edades con menor ocurrencia.